src/ray/core_worker/task_manager.h

```diff
@@ -502,6 +526,11 @@ class TaskManager : public TaskFinisherInterface, public TaskResubmissionInterfa
   /// Optional shutdown hook to call when pending tasks all finish.
   std::function<void()> shutdown_hook_ GUARDED_BY(mu_) = nullptr;
 
+  /// A task state events buffer initialized managed by the CoreWorker. It will be nullptr
+  /// if TaskEventBuffer is not initialized when recording is turned off (either due to
+  /// config or set-up error.)
+  worker::TaskEventBuffer *task_event_buffer_ = nullptr;
```

Comment:
Consider using a reference to a virtual interface, with a Noop buffer / enabled buffer.

Reply:
Sounds good - i guess i will then have a `Enabled()` virtual method on the interface, which serves as a nullptr check but not exposing raw pointer. Similar to what Sang proposed in another comment.